mdsuite/graph_modules/molecular_graph.py

```diff
@@ -60,167 +71,146 @@ def __init__(
         ----------
         experiment : Experiment
                 Experiment object from which to read.
-        from_smiles : bool
-                Build graphs from a smiles string.
-        from_configuration : bool
-                Build graphs from a configuration.
-        smiles_string : str
-                SMILES string to read
-        species : list
-                List of species to build a graph with.
-        """
-        self.experiment = experiment
-        self.from_smiles = from_smiles
-        self.from_configuration = from_configuration
-
-        self.smiles_string = smiles_string
-        self.species = species
+        molecule_input_data : dict
+                Molecule dictionary to use as reference. The reference component is the
+                most critical part. One can either use a smiles string or a reference
+                dict as demonstrated below.
 
-        self.database = Database(self.experiment.database_path / "database.hdf5")
-
-    def _perform_checks(self):
-        """
-        Check the input for inconsistency.
-
-        Returns
-        -------
-        """
-        pass
+                e.g, the input for EMIM-PF6 ionic liquid would be:
 
-    @staticmethod
-    def get_neighbour_list(positions: tf.Tensor, cell: list = None) -> tf.Tensor:
-        """
-        Generate the neighbour list
+                .. code-block::
 
-        Parameters
-        ----------
-        positions: tf.Tensor
-            Tensor with shape (number_of_configurations, n_atoms, 3)
-            representing the coordinates
-        cell: list
-            If periodic boundary conditions are used, please supply the cell
-            dimensions, e.g. [13.97, 13.97, 13.97]. If the cell is provided
-            minimum image convention will be applied!
-        batch_size: int
-            Has to be evenly divisible by the the number of configurations.
+                   {'emim': {'smiles': 'CCN1C=C[N+](+C1)C', 'amount': 20, "cutoff": 1.7},
+                   'PF6': {'smiles': 'F[P-](F)(F)(F)(F)F', 'amount': 20, "cutoff": 1.7}}
```

Comment:
Are these `mdsuite.Molecules` or still dicts?

Reply:
That needs updating yep.

Reply:
85d3ae8